mbed_targets/mbed_targets.py

```diff
@@ -124,22 +124,27 @@ class UnsupportedMode(ToolsError):
     """The Database Mode is unsupported."""
 
 
-class MbedTargets(Collection):
-    """Base Interface to the Target Database."""
+class MbedTargets(Set):
+    """Base Interface to the Target Database.
 
-    def __init__(self, target_data: Iterable[dict]) -> None:
+    Don't instantiate an MbedTargets directly, this class is intended for subclassing.
+
+    Subclasses are expected to override __init__ to create an iterable of MbedTarget objects and pass it to
+    the parent initialiser via super().__init__.
+    """
+
+    @classmethod
+    def _from_iterable(cls: object, it: Iterable) -> "MbedTargets":
```

Comment:
Nit: I personally prefer `iterable` to `it` since it makes parsing very easy. Though I don't know if `it` is an established convention here?

Reply:
Not really a convention, just how it's defined in the `Set` base class. Might be worth changing it to `iterable` though.